### PR TITLE
fix changelog to handle PR's that have a merge coming from master

### PR DESCRIPTION
**Pull request by erwinmombay**
@mkhatib can you take a look?

see sample of this in https://github.com/ampproject/amphtml/pull/2589

### CODE REVIEW

**Comment by dvoytenko** [comment]
LGTM

File: build-system/tasks/changelog.js [hunk 1]
```diff
@@ -306,8 +306,11 @@ function onGitLogSuccess(gitMetadata, logs) {
   assert(typeof logs == 'string', 'git log should be a string.\n' + logs);
   return commits
     .filter(function(commit) {
-      // filter non Pull request merges
-      return commit.indexOf('Merge pull') == 0;
+      // Filter non Pull request merges and also filter
+      // PR's that have merged from some master as it is usually a merge
+      // from our master making it cyclical. (just screws up the PR api requests
+      // to github)
+      return commit.indexOf('Merge pull') == 0 && !(/\/master$/.test(commit));
```
Comment (mkhatib) [inline build-system/tasks/changelog.js]: I've seen PRs sending merge requests from their own master (#2467, #2519). Anyway you can also check the remote or the org name for a merge? `/ampproject\/master$/`?

Comment (erwinmombay) [inline build-system/tasks/changelog.js]: done.

Comment (erwinmombay) [inline build-system/tasks/changelog.js]: actually ive removed this code and changed the git command to:

`git log canary...some-tag --merges --first-parent`

which will prevent those commits from showing up in the commit list i get

**Comment by mkhatib** [comment]
LGTM

**Pull request merged**